### PR TITLE
PWX-35559: add correlation tracing to nsm filter Stats

### PR DESCRIPTION
**Pull request by alicelyu-px**

### CODE REVIEW

File: api/client/client.go [hunk 1]
```diff
@@ -138,6 +157,41 @@ func unix2HTTP(u *url.URL) {
...
+// shouldRoundTripRetry
+func (c *Client) shouldRoundTripRetry(res *http.Response, err error) bool {
+	if http.ErrHandlerTimeout == err || res == nil {
+		return true
+	}
+	return res.StatusCode == http.StatusRequestTimeout ||
+		res.StatusCode == http.StatusGatewayTimeout
+}
+
+// RoundTrip
```
Comment (ggriffiths) [inline api/client/client.go:169]: is this required for this change?

File: api/client/client.go [hunk 2]
```diff
@@ -30,13 +30,17 @@ func NewClient(host, version, userAgent string) (*Client, error) {
 		return nil, fmt.Errorf("Unable to parse provided url: %v", host)
 	}
 	c := &Client{
+		host:        host,
```
Comment (ggriffiths) [inline api/client/client.go:33]: Is this needed? Seems like some extra files were copied here

File: api/client/client.go (hunk 1 above)
Comment (alicelyu-px) [reply api/client/client.go]: > is this required for this change?

This change was on px master. Looks like openstorage master was not vendored into porx master and this change was mistakenly carried over.. I manually removed this change in a later commit. Thanks for the catch!

**Review by ggriffiths — APPROVED** [review]
lgtm thanks for addressing other concerns!

**Pull request merged**